### ISSUE TITLE
Unable to rename templates.

### ISSUE THREAD

**Issue by pjd** [iocage/iocage#533]
Version 0.9.10 2017/12/22
beast:root:~# iocage rename builder-freebsd10 builder-freebsd10.4
Traceback (most recent call last):
  File "/usr/local/bin/iocage", line 10, in <module>
    sys.exit(cli())
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.6/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/iocage/cli/rename.py", line 39, in cli
    ioc.IOCage(exit_on_error=True, jail=jail).rename(new_name)
  File "/usr/local/lib/python3.6/site-packages/iocage/lib/iocage.py", line 1211, in rename
    self.set(f"host_hostuuid={new_name}", rename=True)
  File "/usr/local/lib/python3.6/site-packages/iocage/lib/iocage.py", line 1418, in set
    iocjson.json_set_value(prop)
  File "/usr/local/lib/python3.6/site-packages/iocage/lib/ioc_json.py", line 723, in json_set_value
    self.json_write(conf)
  File "/usr/local/lib/python3.6/site-packages/iocage/lib/ioc_json.py", line 413, in json_write
    'w') as out:
  File "/usr/local/lib/python3.6/contextlib.py", line 81, in __enter__
    return next(self.gen)
  File "/usr/local/lib/python3.6/site-packages/iocage/lib/ioc_common.py", line 420, in open_atomic
    with tempfile(dir=os.path.dirname(os.path.abspath(filepath))) as tmppath:
  File "/usr/local/lib/python3.6/contextlib.py", line 81, in __enter__
    return next(self.gen)
  File "/usr/local/lib/python3.6/site-packages/iocage/lib/ioc_common.py", line 383, in tempfile
    tf = tmp.NamedTemporaryFile(delete=False, suffix=suffix, dir=dir)
  File "/usr/local/lib/python3.6/tempfile.py", line 549, in NamedTemporaryFile
    (fd, name) = _mkstemp_inner(dir, prefix, suffix, flags, output_type)
  File "/usr/local/lib/python3.6/tempfile.py", line 260, in _mkstemp_inner
    fd = _os.open(file, flags, 0o600)
OSError: [Errno 30] Read-only file system: '/iocage/templates/builder-freebsd10/tmpjppailf4'
Exit 1

**Comment by gronke** [comment]
@pjd can you please provide the output of `mount | grep /iocage/templates/builder-freebsd10`?

**Comment by jsegaert** [comment]
The underlying issue is that templates are mounted as read-only and the `rename` function is trying to change files `config.json` (and fstab) in `/iocage/templates/foo/`

```
/root root@trinity#  iocage list -t
+-----+------+-------+--------------+------------------+
| JID | NAME | STATE |   RELEASE    |       IP4        |
+=====+======+=======+==============+==================+
| -   | foo  | down  | 11.1-RELEASE | 192.168.0.210/24 |
+-----+------+-------+--------------+------------------+

/root root@trinity#  mount | grep foo
tank/iocage/templates/foo on /iocage/templates/foo (zfs, local, read-only, nfsv4acls)
tank/iocage/templates/foo/root on /iocage/templates/foo/root (zfs, local, read-only, nfsv4acls)

/root root@trinity#  zfs list -o name,origin,readonly,mountpoint
NAME                                    ORIGIN                                      RDONLY  MOUNTPOINT
tank                                    -                                              off  /mnt/tank
tank/iocage                             -                                              off  /iocage
tank/iocage/download                    -                                              off  /iocage/download
tank/iocage/download/11.0-RELEASE       -                                              off  /iocage/download/11.0-RELEASE
tank/iocage/download/11.1-RELEASE       -                                              off  /iocage/download/11.1-RELEASE
tank/iocage/images                      -                                              off  /iocage/images
tank/iocage/jails                       -                                              off  /iocage/jails
tank/iocage/log                         -                                              off  /iocage/log
tank/iocage/releases                    -                                              off  /iocage/releases
tank/iocage/releases/11.0-RELEASE       -                                              off  /iocage/releases/11.0-RELEASE
tank/iocage/releases/11.0-RELEASE/root  -                                              off  /iocage/releases/11.0-RELEASE/root
tank/iocage/releases/11.1-RELEASE       -                                              off  /iocage/releases/11.1-RELEASE
tank/iocage/releases/11.1-RELEASE/root  -                                              off  /iocage/releases/11.1-RELEASE/root
tank/iocage/templates                   -                                              off  /iocage/templates
tank/iocage/templates/foo               -                                               on  /iocage/templates/foo
tank/iocage/templates/foo/root          tank/iocage/releases/11.1-RELEASE/root@foo      on  /iocage/templates/foo/root
```
I'll submit a PR unless anybody has objections.

**Comment by skarekrow** [comment]
@jsegaert Did this ever get fixed by one of your PR's? I forget

**Comment by jsegaert** [comment]
I have it fixed locally, but haven't submitted the PR yet.  Will do shortly.

**Comment by skarekrow** [comment]
Closed by #570
